toolchain/check/handle_modifier.cpp

```diff
@@ -34,6 +34,9 @@ static auto DiagnoseNotAllowedWith(Context& context, Parse::NodeId first_node,
       .Emit();
 }
 
+// Handles the main modifier keyword (either alone, such as `private`, or the
+// first in a complex modifier, such as `extern library ...`). If valid, adds it
+// to the modifier set and returns true. Otherwise, diagnoses and returns false.
```

Comment:
This sounds to me like it's saying that in `private abstract fn Foo()`, this function handles the `private` but not the `abstract`, which doesn't seem right. After talking to josh11b I think I understand what this is trying to say; assuming this is correct, I would find it much clearer:

```suggestion
// Handles a modifier keyword, which may be a complete modifier (such as `private`), or the
// first token of a complex modifier (such as `extern library ...`). If valid, adds it
// to the modifier set and returns true. Otherwise, diagnoses and returns false.
```

Reply:
In the `extern library` modifier, `library` is also a keyword. I would think a keyword that helps form a modifier is a "modifier keyword". However, this is not called for the "library" modifier keyword. So I think saying this handles "a modifier keyword" could be misleading.

I've tried a different phrasing, does that work for you?